### PR TITLE
Refactor repo

### PR DESCRIPTION
**Pull request by tanepiper**
Hey @JWO719 This is a bit more work on the same branch as #1 , but I refactored the repo too to get it running as a web application to test stuff within the browser.  I didn't make much of a start except to test the geolocation one and it works great!

### CODE REVIEW

**Comment by niklas-wortmann** [comment]
That's impressive. I need some time to take a look into all of that. I will come back during the day. Thanks a lot for all the work you put into that

File: apps/rxjs-web-demo-e2e/cypress.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
```
Comment (niklas-wortmann) [inline apps/rxjs-web-demo-e2e/cypress.json]: I don't see a benefit in using e2e tests here. I think everything should be testable without a ui. So I'd appreciate if you could remove the demo and e2e folders

File: libs/rxjs-web/src/lib/types/observer.d.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  /**
+   * An array of entries returned from the notification API
+   */
+  entries: ReadonlyArray<T>;
+  /**
+   * The Observer from the notification API
+   */
+  observer: Readonly<O>;
+}
+export type IntersectionNotification = ObserverNotification<
```
Comment (niklas-wortmann) [inline libs/rxjs-web/src/lib/types/observer.d.ts]: from a file structuring point of view I'd prefer having the explicit types near by the operator, so IntersectionNotification should be in the same file as the creation operator

File: package.json [hunk 3]
```diff
@@ -1,128 +1,90 @@
...
-  "prettier": {
-    "semi": true,
-    "singleQuote": true
+    "semantic-release-prepare": "ts-node tools/semantic-release-prepare"
   },
-  "commitlint": {
-    "extends": [
-      "@commitlint/config-conventional"
-    ]
+  "private": true,
```
Comment (niklas-wortmann) [inline package.json:31]: this repository is not private and is very likely to be released within the next weeks. Could you remove that private property? Or am I missing something

File: package.json [hunk 4]
```diff
@@ -1,128 +1,90 @@
...
+  "files": [
+    "dist"
+  ],
+  "author": "Jan-Niklas Wortmann <6104311+JWO719@users.noreply.github.com>",
+  "repository": {
+    "type": "git",
+    "url": ""
+  },
+  "contributors": [
+    {
```
Comment (niklas-wortmann) [inline package.json:81]: I really appreciate everything you implemented so far, but there is more needed than submitting one pr to be listet as contributor. Let's see how things go on, but for now I'd appreciate if you could remove this.

**Review by niklas-wortmann — CHANGES_REQUESTED** [review]
That's an amazing pr. I really like it! I just requested some minor changes

File: package.json (hunk 3 above)
Comment (tanepiper) [reply package.json:31]: I this case the top level is private as it's infra code, but the published library is not - it has its own package.json for publishing to npm

File: apps/rxjs-web-demo-e2e/cypress.json (hunk 1 above)
Comment (tanepiper) [reply apps/rxjs-web-demo-e2e/cypress.json]: Sure can, although it I could suggest a demo site is always good, showing some basic usage (can even be deployed to gihub pages) but if your sure it's not needed I can remove it

Comment (niklas-wortmann) [reply apps/rxjs-web-demo-e2e/cypress.json]: I'd prefer adding it when necessary and for now I can not imagine a proper use case. I just want the repo as slim as possible to make it easy to contribute

**Comment by niklas-wortmann** [comment]
thanks a lot what you did here! It's really amazing and I appreciate it a lot!

**Pull request merged**